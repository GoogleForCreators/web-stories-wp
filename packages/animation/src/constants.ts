/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { _x } from '@googleforcreators/i18n';

/**
 * Internal dependencies
 */
import { AnimationType, ScaleDirection } from './types';

export const BEZIER = {
  linear: 'linear',
  in: 'ease-in',
  out: 'ease-out',
  inOut: 'ease-in-out',
  inQuad: 'cubic-bezier(0.55, 0.085, 0.68, 0.53)',
  outQuad: 'cubic-bezier(0.25, 0.46, 0.45, 0.94)',
  inOutQuad: 'cubic-bezier(0.455, 0.03, 0.515, 0.955)',
  inCubic: 'cubic-bezier(0.55, 0.055, 0.675, 0.19)',
  outCubic: 'cubic-bezier(0.215, 0.61, 0.355, 1)',
  inOutCubic: 'cubic-bezier(0.645, 0.045, 0.355, 1)',
  inQuart: 'cubic-bezier(0.895, 0.03, 0.685, 0.22)',
  outQuart: 'cubic-bezier(0.165, 0.84, 0.44, 1)',
  inOutQuart: 'cubic-bezier(0.77, 0, 0.175, 1)',
  inQuint: 'cubic-bezier(0.755, 0.05, 0.855, 0.06)',
  outQuint: 'cubic-bezier(0.23, 1, 0.32, 1)',
  inOutQuint: 'cubic-bezier(0.86, 0, 0.07, 1)',
  inSine: 'cubic-bezier(0.47, 0, 0.745, 0.715)',
  outSine: 'cubic-bezier(0.39, 0.575, 0.565, 1)',
  inOutSine: 'cubic-bezier(0.445, 0.05, 0.55, 0.95)',
  inExpo: 'cubic-bezier(0.95, 0.05, 0.795, 0.035)',
  outExpo: 'cubic-bezier(0.19, 1, 0.22, 1)',
  inOutExpo: 'cubic-bezier(1, 0, 0, 1)',
  inCirc: 'cubic-bezier(0.6, 0.04, 0.98, 0.335)',
  outCirc: 'cubic-bezier(0.075, 0.82, 0.165, 1)',
  inOutCirc: 'cubic-bezier(0.785, 0.135, 0.15, 0.86)',
  default: 'cubic-bezier(0.4, 0.4, 0.0, 1)',
} as const;

export type BezierType = keyof typeof BEZIER;

export const ANIMATION_EFFECTS = {
  DROP: {
    value: AnimationType.EffectDrop,
    name: _x('Drop', 'animation effect', 'web-stories'),
  },
  FADE_IN: {
    value: AnimationType.EffectFadeIn,
    name: _x('Fade In', 'animation effect', 'web-stories'),
  },
  FLY_IN: {
    value: AnimationType.EffectFlyIn,
    name: _x('Fly In', 'animation effect', 'web-stories'),
  },
  PAN: {
    value: AnimationType.EffectPan,
    name: _x('Pan', 'animation effect', 'web-stories'),
  },
  PULSE: {
    value: AnimationType.EffectPulse,
    name: _x('Pulse', 'animation effect', 'web-stories'),
  },
  TWIRL_IN: {
    value: AnimationType.EffectTwirlIn,
    name: _x('Twirl In', 'animation effect', 'web-stories'),
  },
  WHOOSH_IN: {
    value: AnimationType.EffectWhooshIn,
    name: _x('Whoosh In', 'animation effect', 'web-stories'),
  },
  ZOOM: {
    value: AnimationType.EffectZoom,
    name: _x('Scale', 'animation effect', 'web-stories'),
  },
  ROTATE_IN: {
    value: AnimationType.EffectRotateIn,
    name: _x('Rotate In', 'animation effect', 'web-stories'),
  },
};

export const BACKGROUND_ANIMATION_EFFECTS = {
  ZOOM: {
    value: AnimationType.EffectBackgroundZoom,
    name: _x('Zoom', 'animation effect', 'web-stories'),
  },
  PAN: {
    value: AnimationType.EffectBackgroundPan,
    name: ANIMATION_EFFECTS.PAN.name,
  },
  PAN_AND_ZOOM: {
    value: AnimationType.EffectBackgroundPanAndZoom,
    name: _x('Pan and Zoom', 'animation effect', 'web-stories'),
  },
};

export const ANIMATION_PARTS = {
  BLINK_ON: {
    value: AnimationType.BlinkOn,
    name: _x('Blink On', 'animation effect', 'web-stories'),
  },
  BOUNCE: {
    value: AnimationType.Bounce,
    name: _x('Bounce', 'animation effect', 'web-stories'),
  },
  FADE: {
    value: AnimationType.Fade,
    name: _x('Fade', 'animation effect', 'web-stories'),
  },
  FLIP: {
    value: AnimationType.Flip,
    name: _x('Flip', 'animation effect', 'web-stories'),
  },
  FLOAT_ON: {
    value: AnimationType.FloatOn,
    name: _x('Float On', 'animation effect', 'web-stories'),
  },
  MOVE: {
    value: AnimationType.Move,
    name: _x('Move', 'animation effect', 'web-stories'),
  },
  PULSE: {
    value: AnimationType.Pulse,
    name: _x('Pulse', 'animation effect', 'web-stories'),
  },
  SPIN: {
    value: AnimationType.Spin,
    name: _x('Spin', 'animation effect', 'web-stories'),
  },
  ZOOM: {
    value: AnimationType.Zoom,
    name: _x('Zoom', 'animation effect', 'web-stories'),
  },
};

export const SCALE_DIRECTIONS = {
  IN: [ScaleDirection.ScaleInTopLeft, ScaleDirection.ScaleInBottomRight],
  OUT: [ScaleDirection.ScaleOutTopRight, ScaleDirection.ScaleOutBottomLeft],
} as const;

export const BG_MIN_SCALE = 100;
export const BG_MAX_SCALE = 400;
