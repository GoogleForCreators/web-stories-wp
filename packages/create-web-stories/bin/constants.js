export const PRIVATE_REGISTRY_URL = 'http://localhost:4873/';
export const DEFAULT_PROJECT_NAME = 'web-stories';

// React version currently supported by web-stories.
export const REACT_SUPPORTED_VERSION = '17.0.2';

export const SETUP_TYPES = {
  CRA: 'CRA',
  CUSTOM: 'custom',
};

export const BOILERPLATE_NAMES = {
  DASHBOARD: 'dashboard',
  EDITOR: 'editor',
  EDITOR_DASHBOARD: 'editor-dashboard',
}
