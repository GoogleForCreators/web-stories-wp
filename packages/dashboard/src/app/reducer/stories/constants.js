/*
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export const ACTION_TYPES = {
  CREATING_STORY_FROM_TEMPLATE: 'creating_story_from_template',
  CREATE_STORY_FROM_TEMPLATE_SUCCESS: 'create_story_from_template_success',
  CREATE_STORY_FROM_TEMPLATE_FAILURE: 'create_story_from_template_failure',
  LOADING_STORIES: 'loading_stories',
  FETCH_STORIES_SUCCESS: 'fetch_stories_success',
  FETCH_STORIES_FAILURE: 'fetch_stories_failure',
  UPDATE_STORY: 'update_story',
  UPDATE_STORY_FAILURE: 'update_story_failure',
  TRASH_STORY: 'trash_story',
  TRASH_STORY_FAILURE: 'trash_story_failure',
  DUPLICATE_STORY: 'duplicate_story',
  DUPLICATE_STORY_FAILURE: 'duplicate_story_failure',
};

export const defaultStoriesState = {
  error: {},
  isLoading: false,
  stories: {},
  storiesOrderById: [],
  totalStoriesByStatus: {},
  totalPages: null,
};
