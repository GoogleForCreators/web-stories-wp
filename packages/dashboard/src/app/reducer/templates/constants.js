/*
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const ACTION_TYPES = {
  LOADING_TEMPLATES: 'loading_templates',
  FETCH_TEMPLATES_SUCCESS: 'fetch_templates_success',
  FETCH_TEMPLATES_FAILURE: 'fetch_templates_failure',
  PLACEHOLDER: 'placeholder',
};

export const defaultTemplatesState = {
  allPagesFetched: false,
  error: {},
  isLoading: false,
  templates: {},
  templatesByTag: {},
  templatesOrderById: [],
  totalTemplates: null,
  totalPages: null,
};
