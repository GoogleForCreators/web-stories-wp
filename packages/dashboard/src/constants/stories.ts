/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { __, sprintf, _n } from '@googleforcreators/i18n';

/**
 * Internal dependencies
 */
import type { DashboardStory } from '../types/configProvider';

export const STORY_CONTEXT_MENU_ACTIONS = {
  OPEN_IN_EDITOR: 'open-in-editor-action',
  RENAME: 'rename-action',
  DUPLICATE: 'duplicate-action',
  DELETE: 'delete-story-action',
  COPY_STORY_LINK: 'copy-story-link',
  OPEN_STORY_LINK: 'open-story-link',
  CLOSE: 'close-menu',
};

export const STORY_CONTEXT_MENU_ITEMS = [
  {
    label: __('Open in editor', 'web-stories'),
    value: STORY_CONTEXT_MENU_ACTIONS.OPEN_IN_EDITOR,
    isEnabled: (story: DashboardStory) =>
      Boolean(story?.capabilities?.hasEditAction),
  },
  {
    label: __('Open in new tab', 'web-stories'),
    value: STORY_CONTEXT_MENU_ACTIONS.OPEN_STORY_LINK,
    isEnabled: (story: DashboardStory) => Boolean(story?.previewLink),
  },
  {
    label: __('Copy Story URL', 'web-stories'),
    value: STORY_CONTEXT_MENU_ACTIONS.COPY_STORY_LINK,
    isEnabled: (story: DashboardStory) => Boolean(story?.previewLink),
  },
  {
    label: __('Rename', 'web-stories'),
    value: STORY_CONTEXT_MENU_ACTIONS.RENAME,
    separator: 'top',
    isEnabled: (story: DashboardStory) =>
      Boolean(story?.capabilities?.hasEditAction),
  },
  {
    label: __('Duplicate', 'web-stories'),
    value: STORY_CONTEXT_MENU_ACTIONS.DUPLICATE,
    isEnabled: (story: DashboardStory) =>
      Boolean(story?.capabilities?.hasEditAction),
  },
  {
    label: __('Delete Story', 'web-stories'),
    value: STORY_CONTEXT_MENU_ACTIONS.DELETE,
    separator: 'top',
    isEnabled: (story: DashboardStory) =>
      Boolean(story?.capabilities?.hasDeleteAction),
  },
];

export const SORT_DIRECTION = {
  ASC: 'asc',
  DESC: 'desc',
};

export const STORY_SORT_OPTIONS = {
  NAME: 'title',
  DATE_CREATED: 'date',
  LAST_MODIFIED: 'modified',
  CREATED_BY: 'story_author',
};

export const STORY_SORT_KEYS = {
  orderby: STORY_SORT_OPTIONS,
  order: SORT_DIRECTION,
};

export const ORDER_BY_SORT = {
  [STORY_SORT_OPTIONS.NAME]: SORT_DIRECTION.ASC,
  [STORY_SORT_OPTIONS.DATE_CREATED]: SORT_DIRECTION.DESC,
  [STORY_SORT_OPTIONS.LAST_MODIFIED]: SORT_DIRECTION.DESC,
  [STORY_SORT_OPTIONS.CREATED_BY]: SORT_DIRECTION.ASC,
};

export const STORY_SORT_MENU_ITEMS = [
  {
    label: __('Name', 'web-stories'),
    value: STORY_SORT_OPTIONS.NAME,
  },
  {
    label: __('Date Created', 'web-stories'),
    value: STORY_SORT_OPTIONS.DATE_CREATED,
  },
  {
    label: __('Last Modified', 'web-stories'), // default
    value: STORY_SORT_OPTIONS.LAST_MODIFIED,
  },
  {
    label: __('Created By', 'web-stories'),
    value: STORY_SORT_OPTIONS.CREATED_BY,
  },
];

/**
 * All possible story statuses.
 */
const BASE_STATUSES = {
  DRAFT: 'draft',
  FUTURE: 'future',
  PENDING: 'pending',
  PUBLISH: 'publish',
  PRIVATE: 'private',
};

export const STORY_STATUS = {
  ALL: Object.values(BASE_STATUSES).join(','),
  PUBLISHED_AND_FUTURE: [BASE_STATUSES.PUBLISH, BASE_STATUSES.FUTURE].join(','),
  ...BASE_STATUSES,
};

export const DISPLAY_STATUS = {
  [STORY_STATUS.PUBLISH]: __('Published', 'web-stories'),
  [STORY_STATUS.PENDING]: __('Pending', 'web-stories'),
  [STORY_STATUS.FUTURE]: __('Scheduled', 'web-stories'),
  [STORY_STATUS.DRAFT]: __('Draft', 'web-stories'),
  [STORY_STATUS.PRIVATE]: __('Private', 'web-stories'),
};

export const STORY_STATUSES = [
  {
    label: __('All Stories', 'web-stories'),
    value: STORY_STATUS.ALL,
    status: 'all',
  },
  {
    label: __('Drafts', 'web-stories'),
    value: STORY_STATUS.DRAFT,
    status: STORY_STATUS.DRAFT,
  },
  {
    label: __('Pending', 'web-stories'),
    value: STORY_STATUS.PENDING,
    status: STORY_STATUS.PENDING,
  },
  {
    label: __('Published', 'web-stories'),
    value: STORY_STATUS.PUBLISH,
    status: STORY_STATUS.PUBLISH,
  },
  {
    label: __('Scheduled', 'web-stories'),
    value: STORY_STATUS.FUTURE,
    status: STORY_STATUS.FUTURE,
  },
  {
    label: __('Private', 'web-stories'),
    value: STORY_STATUS.PRIVATE,
    status: STORY_STATUS.PRIVATE,
  },
];

export const STORY_VIEWING_LABELS = {
  [STORY_STATUS.ALL]: (n: number) =>
    sprintf(
      /* translators: %d: number of stories in view */
      _n(
        'Viewing <strong>%d</strong> story',
        'Viewing all <strong>%d</strong> stories',
        n,
        'web-stories'
      ),
      String(n)
    ),
  [STORY_STATUS.DRAFT]: (n: number) =>
    sprintf(
      /* translators: %d: number of draft stories in view */
      _n(
        'Viewing <strong>%d</strong> draft',
        'Viewing <strong>%d</strong> drafts',
        n,
        'web-stories'
      ),
      String(n)
    ),
  [STORY_STATUS.PUBLISH]: (n: number) =>
    sprintf(
      /* translators: %d: number of stories */
      _n(
        'Viewing <strong>%d</strong> published story',
        'Viewing <strong>%d</strong> published stories',
        n,
        'web-stories'
      ),
      String(n)
    ),
  [STORY_STATUS.FUTURE]: (n: number) =>
    sprintf(
      /* translators: %d: number of stories */
      _n(
        'Viewing <strong>%d</strong> scheduled story',
        'Viewing <strong>%d</strong> scheduled stories',
        n,
        'web-stories'
      ),
      String(n)
    ),
  [STORY_STATUS.PENDING]: (n: number) =>
    sprintf(
      /* translators: %d: number of stories */
      _n(
        'Viewing <strong>%d</strong> pending story',
        'Viewing <strong>%d</strong> pending stories',
        n,
        'web-stories'
      ),
      String(n)
    ),
  [STORY_STATUS.PRIVATE]: (n: number) =>
    sprintf(
      /* translators: %d: number of stories */
      _n(
        'Viewing <strong>%d</strong> private story',
        'Viewing <strong>%d</strong> private stories',
        n,
        'web-stories'
      ),
      String(n)
    ),
};

export const STORY_ANIMATION_STATE = {
  RESET: 'reset',
  PAUSED: 'paused',
  SCRUBBING: 'scrubbing',
  PLAYING: 'playing',
};

export const DEFAULT_FILTERS = {
  filters: {
    status: STORY_STATUS.ALL,
  },
  sort: {
    orderby: STORY_SORT_OPTIONS.LAST_MODIFIED,
    order: ORDER_BY_SORT[STORY_SORT_OPTIONS.LAST_MODIFIED],
  },
};
