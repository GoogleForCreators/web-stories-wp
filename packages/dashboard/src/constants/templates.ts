/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { __, _x, sprintf, _n } from '@googleforcreators/i18n';

export const TEMPLATE_CATEGORIES = {
  DIY_CRAFTS: 'diy_crafts',
  MOVIES_TV: 'movies_tv',
  BEAUTY_STYLE: 'beauty_style',
  FITNESS_WELLBEING: 'fitness_wellbeing',
  FOOD: 'food',
  TRAVEL: 'travel',
  MUSIC: 'music',
  SPORTS: 'sports',
};

export const TEMPLATE_CATEGORY_ITEMS = [
  {
    label: _x('DIY & Crafts', 'template vertical', 'web-stories'),
    value: TEMPLATE_CATEGORIES.DIY_CRAFTS,
  },
  {
    label: _x('Movies & TV', 'template vertical', 'web-stories'),
    value: TEMPLATE_CATEGORIES.MOVIES_TV,
  },
  {
    label: _x('Beauty & Style', 'template vertical', 'web-stories'),
    value: TEMPLATE_CATEGORIES.BEAUTY_STYLE,
  },
  {
    label: _x('Fitness & Wellbeing', 'template vertical', 'web-stories'),
    value: TEMPLATE_CATEGORIES.FITNESS_WELLBEING,
  },
  {
    label: _x('Food', 'template vertical', 'web-stories'),
    value: TEMPLATE_CATEGORIES.FOOD,
  },
  {
    label: _x('Travel', 'template vertical', 'web-stories'),
    value: TEMPLATE_CATEGORIES.TRAVEL,
  },
  {
    label: _x('Music', 'template vertical', 'web-stories'),
    value: TEMPLATE_CATEGORIES.MUSIC,
  },
  {
    label: _x('Sports', 'template vertical', 'web-stories'),
    value: TEMPLATE_CATEGORIES.SPORTS,
  },
];

export const TEMPLATE_COLORS = {
  WHITE: 'white',
  BLACK: 'black',
  GRAY: 'gray',
  BROWN: 'brown',
  RED: 'red',
  ORANGE: 'orange',
  YELLOW: 'yellow',
  GREEN: 'green',
  BLUE: 'blue',
  PINK: 'pink',
  PURPLE: 'purple',
};

export const TEMPLATE_COLOR_ITEMS = [
  {
    label: _x('White', 'color', 'web-stories'),
    hex: '#FFFFFF',
    value: TEMPLATE_COLORS.WHITE,
  },
  {
    label: _x('Black', 'color', 'web-stories'),
    hex: '#1A1D1F',
    value: TEMPLATE_COLORS.BLACK,
  },
  {
    label: _x('Gray', 'color', 'web-stories'),
    hex: '#9AA1A9',
    value: TEMPLATE_COLORS.GRAY,
  },
  {
    label: _x('Brown', 'color', 'web-stories'),
    hex: '#974A04',
    value: TEMPLATE_COLORS.BROWN,
  },
  {
    label: _x('Red', 'color', 'web-stories'),
    hex: '#EB5757',
    value: TEMPLATE_COLORS.RED,
  },
  {
    label: _x('Orange', 'color', 'web-stories'),
    hex: '#FA902E',
    value: TEMPLATE_COLORS.ORANGE,
  },
  {
    label: _x('Yellow', 'color', 'web-stories'),
    hex: '#FAE84C',
    value: TEMPLATE_COLORS.YELLOW,
  },
  {
    label: _x('Green', 'color', 'web-stories'),
    hex: '#6AE86F',
    value: TEMPLATE_COLORS.GREEN,
  },
  {
    label: _x('Blue', 'color', 'web-stories'),
    hex: '#1374FA',
    value: TEMPLATE_COLORS.BLUE,
  },
  {
    label: _x('Pink', 'color', 'web-stories'),
    hex: '#F078F2',
    value: TEMPLATE_COLORS.PINK,
  },
  {
    label: _x('Purple', 'color', 'web-stories'),
    hex: '#AA6FE1',
    value: TEMPLATE_COLORS.PURPLE,
  },
];

export const TEMPLATES_GALLERY_STATUS = {
  ALL: 'template',
};

export const TEMPLATES_GALLERY_VIEWING_LABELS = {
  [TEMPLATES_GALLERY_STATUS.ALL]: (n: number) =>
    sprintf(
      /* translators: %d: number of templates in view */
      _n(
        'Viewing <strong>%d</strong> template',
        'Viewing all <strong>%d</strong> templates',
        n,
        'web-stories'
      ),
      String(n)
    ),
};

export const TEMPLATE_SORT_DIRECTION = {
  ASC: 'asc',
  DESC: 'desc',
};

export const TEMPLATES_GALLERY_SORT_OPTIONS = {
  POPULAR: 'popular',
  RECENT: 'recent',
};

export const TEMPLATE_SORT_KEYS = {
  orderby: TEMPLATES_GALLERY_SORT_OPTIONS,
  order: TEMPLATE_SORT_DIRECTION,
};

export const DEFAULT_TEMPLATE_FILTERS = {
  filters: {
    status: TEMPLATES_GALLERY_STATUS.ALL,
  },
  sort: {
    orderby: TEMPLATES_GALLERY_SORT_OPTIONS.POPULAR,
  },
};

export const TEMPLATES_GALLERY_SORT_MENU_ITEMS = [
  {
    label: __('Popular', 'web-stories'),
    value: TEMPLATES_GALLERY_SORT_OPTIONS.POPULAR,
  },
  {
    label: __('Recent', 'web-stories'),
    value: TEMPLATES_GALLERY_SORT_OPTIONS.RECENT,
  },
];

export const TEMPLATES_GALLERY_ITEM_CENTER_ACTION_LABELS = {
  template: __('See details', 'web-stories'),
};
