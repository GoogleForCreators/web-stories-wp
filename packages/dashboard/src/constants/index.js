/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { __ } from '@googleforcreators/i18n';
/**
 * Internal dependencies
 */
import { Compass, HomeWithHeart } from '../icons';
import { STORY_VIEWING_LABELS } from './stories';
import { TEMPLATES_GALLERY_VIEWING_LABELS } from './templates';

export const KEYBOARD_USER_CLASS = `useskeyboard`;

export const Z_INDEX = {
  LAYOUT_SQUISHABLE: 3,
  LAYOUT_FIXED: 5,
  STICKY_TABLE: 2,
  POPOVER_MENU: 10,
  POPOVER_PANEL: 10,
};

export const APP_ROUTES = {
  DASHBOARD: '/',
  TEMPLATES_GALLERY: '/templates-gallery',
  TEMPLATE_DETAIL: 'template-detail',
};

export const ROUTE_TITLES = {
  [APP_ROUTES.DASHBOARD]: __('Dashboard', 'web-stories'),
  [APP_ROUTES.TEMPLATES_GALLERY]: __('Explore Templates', 'web-stories'),
  [`${APP_ROUTES.TEMPLATES_GALLERY}/${APP_ROUTES.TEMPLATE_DETAIL}`]: __(
    'Template Details',
    'web-stories'
  ),
  DEFAULT: __('Dashboard', 'web-stories'),
};

export const PRIMARY_PATHS = [
  {
    value: APP_ROUTES.DASHBOARD,
    label: ROUTE_TITLES[APP_ROUTES.DASHBOARD],
    Icon: HomeWithHeart,
  },
  {
    value: APP_ROUTES.TEMPLATES_GALLERY,
    label: ROUTE_TITLES[APP_ROUTES.TEMPLATES_GALLERY],
    Icon: Compass,
  },
];

export const VIEW_STYLE = {
  GRID: 'grid',
  LIST: 'list',
};

export const VIEW_STYLE_LABELS = {
  [VIEW_STYLE.GRID]: __('Switch to List View', 'web-stories'),
  [VIEW_STYLE.LIST]: __('Switch to Grid View', 'web-stories'),
};

export const DASHBOARD_VIEWS = {
  DASHBOARD: 'DASHBOARD',
  TEMPLATES_GALLERY: 'TEMPLATES_GALLERY',
};
export const RESULT_LABELS = {
  [DASHBOARD_VIEWS.DASHBOARD]: { ...STORY_VIEWING_LABELS },
  [DASHBOARD_VIEWS.TEMPLATES_GALLERY]: { ...TEMPLATES_GALLERY_VIEWING_LABELS },
};

// API Query Constants
export const STORIES_PER_REQUEST = 24;

export const TEXT_INPUT_DEBOUNCE = 300;

export const MIN_IMG_HEIGHT = 96;
export const MIN_IMG_WIDTH = 96;

export const DEFAULT_GRID_IMG_HEIGHT = 853;
export const DEFAULT_GRID_IMG_WIDTH = 640;

export * from './components';
export * from './pageStructure';
export * from './stories';
export * from './templates';
