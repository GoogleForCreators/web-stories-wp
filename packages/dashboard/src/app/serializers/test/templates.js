/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * External dependencies
 */
import { getOptions, toDate, toUTCDate } from '@web-stories-wp/date';

/**
 * Internal dependencies
 */
import { APP_ROUTES } from '../../../constants';
import reshapeTemplateObject from '../templates';

describe('reshapeTemplateObject', () => {
  const templateData = {
    id: 1,
    slug: 'beauty',
    title: 'Beauty',
    createdBy: 'Google',
    creationDate: '2020-03-01T07:00:00.000Z',
    modified: '2020-04-21T07:00:00.000Z',
    tags: ['Health', 'Bold', 'Joy'],
    colors: [
      { label: 'Pink', color: '#f3d9e1' },
      { label: 'Green', color: '#d8ddcc' },
      { label: 'Black', color: '#28292b' },
    ],
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    pages: [{ id: 0, elements: [] }],
    version: 17,
  };

  it('should return object', () => {
    const responseObj = {
      id: 1,
      slug: 'template-1-slug',
      pages: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }],
      creationDate: '2020-03-26T20:57:24',
      modified: '2020-03-26T20:57:24',
    };

    const reshapedObj = reshapeTemplateObject(responseObj, 'example.com/');

    expect(reshapedObj).toMatchObject({
      id: 1,
      slug: 'template-1-slug',
      postersByPage: {
        0: {
          webp: 'example.com/images/templates/template-1-slug/posters/1.webp',
          png: 'example.com/images/templates/template-1-slug/posters/1.png',
        },
        1: {
          webp: 'example.com/images/templates/template-1-slug/posters/2.webp',
          png: 'example.com/images/templates/template-1-slug/posters/2.png',
        },
        2: {
          webp: 'example.com/images/templates/template-1-slug/posters/3.webp',
          png: 'example.com/images/templates/template-1-slug/posters/3.png',
        },
      },
      centerTargetAction: `${APP_ROUTES.TEMPLATE_DETAIL}?id=1&isLocal=false`,
      creationDate: toDate('2020-03-26T20:57:24.000Z'),
      status: 'template',
      modified: toDate('2020-03-26T20:57:24.000Z'),
    });
  });

  it('should not return an object', () => {
    const reshapedObj = reshapeTemplateObject(
      {
        pages: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }],
        creationDate: '2020-03-26T20:57:24',
      },
      'example.com/'
    );

    expect(reshapedObj).toBeNull();
  });

  it('should reshape object to match snapshot', () => {
    const reshapedObject = reshapeTemplateObject(templateData, 'example.com/');
    expect(reshapedObject).toMatchSnapshot();
  });

  it('should reshape template object with a date object', () => {
    const reshapedObject = reshapeTemplateObject(templateData, 'example.com/');
    expect(reshapedObject.modified).toMatchObject(
      toUTCDate('2020-04-21T07:00:00.000Z')
    );

    expect(reshapedObject.creationDate).toMatchObject(
      toDate('2020-03-01T07:00:00.000Z', getOptions())
    );
  });

  it('should pass through tags and colors into reshaped object', () => {
    const reshapedObject = reshapeTemplateObject(templateData, 'example.com/');

    expect(reshapedObject.tags).toMatchObject(['Health', 'Bold', 'Joy']);
    expect(reshapedObject.colors).toMatchObject([
      { label: 'Pink', color: '#f3d9e1' },
      { label: 'Green', color: '#d8ddcc' },
      { label: 'Black', color: '#28292b' },
    ]);
  });

  it('should apply isLocal to the reshaped object', () => {
    const localReshapedObject = reshapeTemplateObject(
      templateData,
      'example.com/',
      true
    );
    expect(localReshapedObject.isLocal).toBe(true);

    const notlocalReshapedObject = reshapeTemplateObject(
      templateData,
      'example.com/'
    );
    expect(notlocalReshapedObject.isLocal).toBe(false);
  });
});
