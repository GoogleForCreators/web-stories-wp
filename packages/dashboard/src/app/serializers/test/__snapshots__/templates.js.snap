// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reshapeTemplateObject should reshape object to match snapshot 1`] = `
Object {
  "centerTargetAction": "template-detail?id=1&isLocal=false",
  "colors": Array [
    Object {
      "color": "#f3d9e1",
      "label": "Pink",
    },
    Object {
      "color": "#d8ddcc",
      "label": "Green",
    },
    Object {
      "color": "#28292b",
      "label": "Black",
    },
  ],
  "createdBy": "Google",
  "creationDate": 2020-03-01T07:00:00.000Z,
  "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
  "id": 1,
  "isLocal": false,
  "modified": 2020-04-21T07:00:00.000Z,
  "pages": Array [
    Object {
      "elements": Array [],
      "id": 0,
    },
  ],
  "postersByPage": Object {
    "0": Object {
      "png": "example.com/images/templates/beauty/posters/1.png",
      "webp": "example.com/images/templates/beauty/posters/1.webp",
    },
  },
  "slug": "beauty",
  "status": "template",
  "tags": Array [
    "Health",
    "Bold",
    "Joy",
  ],
  "title": "Beauty",
  "version": 17,
}
`;
